fractal_server/app/runner/_slurm/executor.py

```diff
@@ -0,0 +1,228 @@
+# Copyright 2021 Adrian Sampson <asampson@cs.washington.edu>
+# License: MIT
+#
+# Copyright 2022 Jacopo Nepsolo <jacopo.nespolo@exact-lab.it>
+import shlex
+import subprocess  # nosec
+import sys
+from concurrent import futures
+from pathlib import Path
+from typing import Any
+from typing import Callable
+from typing import List
+from typing import Optional
+
+import cloudpickle
+from cfut import RemoteException
+from cfut import SlurmExecutor  # type: ignore
+from cfut.util import random_string
+
+from ....config import get_settings
+from ....syringe import Inject
+from ....utils import set_logger
+
+
+def get_slurm_script_dir() -> Path:
+    settings = Inject(get_settings)
+    script_dir = settings.RUNNER_ROOT_DIR  # type: ignore
+    return script_dir  # type: ignore
+
+
+def get_stdout_filename(arg: str = "%j") -> Path:
+    return get_slurm_script_dir() / f"slurmpy.stdout.{arg}.log"
+
+
+def get_in_filename(arg) -> Path:
+    return get_slurm_script_dir() / f"cfut.in.{arg}.pickle"
+
+
+def get_out_filename(arg) -> Path:
+    return get_slurm_script_dir() / f"cfut.out.{arg}.pickle"
+
+
+def write_batch_script(
+    sbatch_script: str, script_dir: Optional[Path] = None
+) -> Path:
+    """
+    Write batch script
+
+    Returns:
+        batch_script_path:
+            The path to the batch script
+    """
+    if not script_dir:
+        script_dir = get_slurm_script_dir()
+
+    batch_script_path = script_dir / f"_temp_{random_string()}.sh"
+    with batch_script_path.open("w") as f:
+        f.write(sbatch_script)
+    return batch_script_path
+
+
+def submit_sbatch(
+    sbatch_script: str,
+    submit_pre_command: str = "",
+    script_dir: Optional[Path] = None,
+) -> int:
+    """
+    Submit a Slurm job script
+
+    Write the batch script in a temporary file and submit it with `sbatch`.
+
+    Args:
+        sbatch_script:
+            the string representing the full job
+        submit_pre_command:
+            command that is prefixed to `sbatch`
+        script_dir:
+            destination of temporary script files
+
+    Returns:
+        jobid:
+            integer job id as returned by `sbatch` submission
+    """
+    filename = write_batch_script(
+        sbatch_script=sbatch_script, script_dir=script_dir
+    )
+    submit_command = f"sbatch --parsable {filename}"
+    full_cmd = shlex.join(
+        shlex.split(submit_pre_command) + shlex.split(submit_command)
+    )
+    try:
+        output = subprocess.run(  # nosec
+            full_cmd, capture_output=True, check=True
+        )
+    except subprocess.CalledProcessError as e:
+        logger = set_logger(logger_name="slurm_runner")
+        logger.error(e.stderr)
+        raise e
+    jobid = output.stdout
+    # NOTE after debugging this can be uncommented
+    # filename.unlink()
+    return int(jobid)
+
+
+def compose_sbatch_script(
+    cmdline: List[str],
+    # NOTE: In SLURM, `%j` is the placeholder for the job_id.
+    outpat: Optional[Path] = None,
```

Comment:
typo (outpat->output or outpat->outpath) throughout the function

Reply:
it is intentional, as it is the original variable name from upstream.